backend/api/Areas/Insurances/Mapping/InsuranceMap.cs

```diff
@@ -0,0 +1,32 @@
+using Mapster;
+using Entity = Pims.Dal.Entities;
+using Model = Pims.Api.Areas.Insurance.Models;
+
+namespace Pims.Api.Areas.Insurance.Mapping
+{
+    public class InsuranceMap : IRegister
+    {
+        public void Register(TypeAdapterConfig config)
+        {
+            config.NewConfig<Entity.PimsInsurance, Model.InsuranceModel>()
+                .Map(dest => dest.Id, src => src.InsuranceId)
+                .Map(dest => dest.InsuranceType, src => src.InsuranceTypeCodeNavigation)
+                .Map(dest => dest.OtherInsuranceType, src => src.OtherInsuranceType)
+                .Map(dest => dest.CoverageDescription, src => src.CoverageDescription)
+                .Map(dest => dest.CoverageLimit, src => src.CoverageLimit)
+                .Map(dest => dest.ExpiryDate, src => src.ExpiryDate)
+                .Map(dest => dest.IsInsuranceInPlace, src => src.IsInsuranceInPlace)
+                .Inherits<Entity.IBaseEntity, Api.Models.BaseModel>();
+
+            config.NewConfig<Model.InsuranceModel, Entity.PimsInsurance>()
+                .Map(dest => dest.InsuranceId, src => src.Id)
+                .Map(dest => dest.InsuranceTypeCode, src => src.InsuranceType.Id)
```

Comment:
There's potential for NPE here as we don't control the JSON sent to the API. It could have `src.InsuranceType` be undefined/null. Should probably check for null before trying to do `src.InsuranceType.Id` ?

Reply:
I think if the insurance does not have a type an exception should be valid since that is should never be null